Automattic/jetpack
# Admin Page: Move to Core translation functions -- PART 1

#### Changes proposed in this Pull Request:

**This will need to happen in steps so reviews are easier. This is step 1.**

We're aiming to move away from the `i18n-calypso` tooling to handle translations. Instead we'll rely on 2 things:
- `@wordpress/i18n` for most translation work: `__`, `_n`, `_x`, and `sprintf`.
- `@wordpress/element` and its `createInterpolateElement` to handle React elements in strings.

`i18n-calypso` will remain in use for a few exceptions for now:
- `moment`
- `numberFormat`

This first PR introduces those changes for the At A Glance dashboard.

#### Jetpack product discussion

* p1HpG7-99c-p2
* Primary issue: #16481 

#### Does this pull request change what data or activity we track or use?

* N/A

#### Testing instructions:

Nothing should change at first sight:

- Under Jetpack > Dashboard, check how things look like everywhere on the page when you've connected your site to WordPress.com, or when Development mode is active.
- Try with different plugins and module active / inactive to get different card statues.
- Try with outdated plugins to see different messages.
- If you view source on the page, you should see a new section right before the React JS bundle is loaded: `react-plugin-js-translations`. There are no translations here for now, as no language pack is generated yet for this JS bundle. We'll need to push an Alpha first.

#### Proposed changelog entry for your changes:

* TBD


******************************

#### Original PR description below (before the discussion in the comments below happened)

This is a demo PR that shows how we'll be able to remove all of our uses of `translate()` (from i18n-calypso) within the Jetpack Dashboard so that we can move over to using language pack JSONs for this.

This could be merged as is and would have the Akismet "At a glance" use translations loaded the standard WordPress/Gutenberg way (i.e. using `wp_set_script_translations()`). The downside of this is that this duplicates the translations JSON in the HTML source.

To bring this over the finish line we'll also need to change our build process so that the translatable strings are picked up from the JavaScript source. We currently only ship the uglified `_inc/build/admin.js` and not the JavaScript clean source. If we committed the unmodified source to SVN, the strings would be extracted for translation and provided inside language packs.

To avoid duplication, the [convention in building the language packs](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/wporg-gp-customizations/inc/cli/class-language-pack.php#L441) is to have the source be placed under a `src/` path, so that they won't be included in the language pack.
